From 0e0c19ebcda2aa89144e724e8395e514452721ee Mon Sep 17 00:00:00 2001
From: Shawn8901 <shawn8901@googlemail.com>
Date: Wed, 30 Mar 2022 19:38:28 +0200
Subject: [PATCH] Adds a cmake option to control if a placeholder is written on
 install

This option enables the possibility for a package maintainer to set RTTR_GAMEDIR to a directory outside the package itself and do no write operation.
That is us useful for systems where the packaging system prevents write operations outside of the package structure on package build time. A system where this is the case is NixOS.

As the use case is special for some packaging systems and not intended for daily use it is marked as an advanced option.
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4562487ef..688d873bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,7 +384,11 @@ if(NOT WIN32)
 endif()
 
 # Placeholder for S2 installation
-install(FILES "${RTTR_S2_PLACEHOLDER_PATH}" DESTINATION "${RTTR_GAMEDIR}")
+option(RTTR_INSTALL_PLACEHOLDER "Install a placeholder file to the location the S2 game files should be copied to." ON)
+mark_as_advanced(RTTR_INSTALL_PLACEHOLDER)
+if(RTTR_INSTALL_PLACEHOLDER)
+    install(FILES "${RTTR_S2_PLACEHOLDER_PATH}" DESTINATION "${RTTR_GAMEDIR}")
+endif()
 
 ################################################################################
 # Postbuild
